app/Actions/Search/AlbumSearch.php

```diff
@@ -2,36 +2,59 @@
 
 namespace App\Actions\Search;
 
-use App\Actions\Albums\Extensions\PublicIds;
-use App\Facades\AccessControl;
+use App\Actions\AlbumAuthorisationProvider;
 use App\Models\Album;
+use App\Models\TagAlbum;
 use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Collection;
 
 class AlbumSearch
 {
-	public function query(array $terms)
+	protected AlbumAuthorisationProvider $albumAuthorisationProvider;
+
+	public function __construct()
+	{
+		$this->albumAuthorisationProvider = resolve(AlbumAuthorisationProvider::class);
```

Comment:
Unless I am wrong, this should be using DDI.

Reply:
Fixed it.